src/main/java/ama/post/PostController.java

```diff
@@ -94,22 +93,31 @@ public String likeSubmissionPost(HttpServletRequest request, @RequestParam(value
     public String upvoteComment(HttpServletRequest request, @RequestParam("id") String commentId) {
         CommentPost comment = commentPostRepository.findById(Long.parseLong(commentId));
         User user = authenticator.getCurrentUser();
-        if (user != null && comment != null) {
-            comment.upvote(user);
-            commentPostRepository.save(comment);
+        if (comment != null) {
+            SubmissionPost post = comment.getSubmissionPost();
+            Date now = new Date();
+            if (user != null && now.before(post.getVotingCloses())) {
+                comment.upvote(user);
+                commentPostRepository.save(comment);
+            }
             String referer = request.getHeader("Referer");
             return "redirect:" + referer;
+        } else {
+            return "pageNotFound";
```

Comment:
why redirect to not found for upvote but not downvote? I think we shouldn't show it for either